latok/core/latok_utils.py

```diff
@@ -114,3 +119,117 @@ def feature_weights(self):
         Get the non-zero feature names mapped to their weights.
         '''
         return {FEATURE_NAMES[idx]: self.features[idx] for idx in np.nonzero(self.features)[0]}
+
+    def add_abstract_feature(self, feature_name):
+        if self.abstract_features is None:
+            self.abstract_features = list()
+        self.abstract_features.append(feature_name)
+
+
+def _apply_combo_matrix(combo_matrix, mt):
+    s = _combine_matrix_rows(mt, combo_matrix)
+    return np.nonzero(s)[0]
+
+
+class OffsetSpec:
+    '''
```

Comment:
I'm not sure I understand this class and its purpose

Reply:
See comments below.